sunpy/coordinates/wcs_utils.py

```diff
@@ -32,17 +37,31 @@ def solar_wcs_frame_mapping(wcs):
 
     dateobs = wcs.wcs.dateobs or None
 
-    # SunPy Map adds 'heliographic_observer' and 'rsun' attributes to the WCS
-    # object. We check for them here, and default to None.
+    # These custom attributes were used in sunpy < 2.1; these warnings can
+    # be removed in sunpy 3.1
     if hasattr(wcs, 'heliographic_observer'):
-        observer = wcs.heliographic_observer
-    else:
-        observer = None
-
+        warnings.warn('sunpy no longer uses the .heliographic_observer attribute.',
```

Comment:
If this isn't going to work, should this be an error? Also, it should point people at aux, so they know how to fix it.

Reply:
Dunno, having these attributes isn't going to do any harm, they just won't be used. I was thinking this was sort of a deprecation period, after which we can just remove the warnings.

Reply:
The normal flow for a deprecation is that the thing still works but shouts at you, and then it is removed. Here we have it raise a warning and then not work. I think we should either keep it working or make it raise an error, as you need to edit your code to get the expected behaviour.

Reply:
I've put back support for using the old method, with deprecation warnings. Still need to add tests for this and update the changelog.

Reply:
Updated changelog and should be tested too.